File: hu/codes/python/chapter_stack_and_queue/queue.py
"""
File: queue.py
Created Time: 2022-11-29
Author: Peng Chen (pengchzn@gmail.com)
"""

from collections import deque

"""Fő kód"""
if __name__ == "__main__":
    # Sor inicializálása
    # Pythonban általában a kétirányú sor (deque) osztályt használjuk sorként
    # Bár a queue.Queue() egy megfelelő sor osztály, nem túl kényelmes használni
    que: deque[int] = deque()

    # Elemek sorba állítása
    que.append(1)
    que.append(3)
    que.append(2)
    que.append(5)
    que.append(4)
    print("sor que =", que)

    # A sor elején lévő elem elérése
    front: int = que[0]
    print("Sor eleje front =", front)

    # Elem kivétele a sorból
    pop: int = que.popleft()
    print("Kivett elem pop =", pop)
    print("Kivétel után que =", que)

    # A sor hosszának lekérése
    size: int = len(que)
    print("Sor hossza size =", size)

    # Ellenőrzi, hogy a sor üres-e
    is_empty: bool = len(que) == 0
    print("A sor üres-e =", is_empty)

File: hu/codes/java/chapter_stack_and_queue/queue.java
/**
 * File: queue.java
 * Created Time: 2022-11-25
 * Author: krahets (krahets@163.com)
 */

package chapter_stack_and_queue;

import java.util.*;

public class queue {
    public static void main(String[] args) {
        /* Sor elejének elérése */
        Queue<Integer> queue = new LinkedList<>();

        /* Elemek sorba rakása */
        queue.offer(1);
        queue.offer(3);
        queue.offer(2);
        queue.offer(5);
        queue.offer(4);
        System.out.println("Queue queue = " + queue);

        /* Lista visszaadása nyomtatáshoz */
        int peek = queue.peek();
        System.out.println("Front element peek = " + peek);

        /* Elem kivétele a sorból */
        int pop = queue.poll();
        System.out.println("Dequeue element pop = " + pop + ", after dequeue, queue = " + queue);

        /* Sor hosszának lekérése */
        int size = queue.size();
        System.out.println("Queue length size = " + size);

        /* Sor üresség ellenőrzése */
        boolean isEmpty = queue.isEmpty();
        System.out.println("Queue is empty = " + isEmpty);
    }
}
